src/main/java/novamachina/exnihilosequentia/common/tileentity/barrel/AbstractBarrelTile.java

```diff
@@ -52,6 +54,35 @@
     private final LazyOptional<IFluidHandler> tankHolder = LazyOptional.of(() -> tank);
     private AbstractBarrelMode mode;
     private int solidAmount;
+    private AbstractBarrelTileState lastSyncedState = null;
+
+    static protected class AbstractBarrelTileState {
+        private final AbstractBarrelMode mode;
+        private final Fluid fluid;
+        private final int fluidAmount;
+        private final Item solid;
+        private final int solidAmount;
+
+        AbstractBarrelTileState (final AbstractBarrelTile abstractBarrelTile) {
+            mode = abstractBarrelTile.getMode();
+            fluid = abstractBarrelTile.getFluid();
+            fluidAmount = abstractBarrelTile.getFluidAmount();
+            solid = abstractBarrelTile.inventory.getStackInSlot(0).getItem();
+            solidAmount = abstractBarrelTile.getSolidAmount();
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            AbstractBarrelTileState that = (AbstractBarrelTileState) o;
+            return fluidAmount == that.fluidAmount
+                    && solidAmount == that.solidAmount
+                    && Objects.equals(mode, that.mode)
```

Comment:
One other thing that I noticed while testing was that HWYLA no longer updates because the state is not getting synced to the client as often. I believe this is an issue with this mode comparison. Since the modes do not have an equals function, this does not appear to sync when the mode values update.

Reply:
Good spot, had completely missed this. The modeName doesn't look like it can be relied on to be different, so am encapsulating the wailaInfo instead now. Hopefully this makes it less likely to break with any future changes.